### ISSUE TITLE
[BUG] `ReactionSet.add_rxn_set()` failing falsely due to unlike entries (specifically CO2)

### ISSUE THREAD

**Issue by mattmcdermott** [materialsproject/reaction-network#285]
An error like this has been arising when trying to add a reaction set to another.
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[34], line 1
----> 1 rxns1.add_rxn_set(rxns2)

File ~/code/github/reaction-network/src/rxn_network/reactions/reaction_set.py:310, in ReactionSet.add_rxn_set(self, rxn_set)
    303 """Adds a new reaction set to current reaction set.
    304 
    305 Warning: new reaction set must have the same entries as the current reaction
    306 set.
    307 
    308 """
    309 if self.entries != rxn_set.entries:
--> 310     raise ValueError("Reaction sets must have identical entries property to combine.")
    312 open_elem = self.open_elem
    313 chempot = self.chempot

ValueError: Reaction sets must have identical entries property to combine.
```

This is related to the CO2 entries being different and is likely due to the addition of the atmospheric CO2 correction.

Will address ASAP

**Comment by mattmcdermott** [comment]
This is actually a general problem with comparing energy adjustments. I have updated the `ExperimentalReferenceEntry` class to include a check for the value of the energy adjustment rather than the actual adjustment itself.

**Comment by mattmcdermott** [comment]
Fixed with merge of #286
